WalletWasabi/Blockchain/TransactionBuilding/BranchAndBound.cs

```diff
@@ -2,18 +2,23 @@
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using WalletWasabi.Logging;
+using System.Threading;
+using WalletWasabi.Blockchain.TransactionBuilding.BnB;
 
 namespace WalletWasabi.Blockchain.TransactionBuilding;
 
 /// <seealso href="https://murch.one/wp-content/uploads/2016/11/erhardt2016coinselection.pdf">Section "5.3 Branch and Bound".</seealso>
 public class BranchAndBound
 {
 	private readonly Random _random = new();
+	private readonly Func<bool> _randomFn;
 
 	/// <param name="values">All values must be strictly positive.</param>
-	public BranchAndBound(List<long> values)
+	/// <param name="randomFn">Function returning randomly <c>true</c> and <c>false</c>.</param>
+	public BranchAndBound(List<long> values, Func<bool>? randomFn = null)
```

Comment:
Why is `randomFn` a param? Isn't just LOC20 enough?
`_randomFn = () => _random.Next(0, 2) == 1;`

Reply:
The param is mainly for testing. It is useful for testing edge cases like: `When BnB algorithm runs, we always choose NOT to include a coin first.` (an example would be: you want to put together 500 sats and you have 499, 1,1,1,1,1,1,1 ... then the algo we skip 499 and attempts to put together 500 from a few 1s which will fail so many CPU cycles are actually wasted... something like this.)

We can have a second constructor for this marked with some comment that it is only for testing. 

It is very important to understand that randomness here can be in your favor or against you really by choosing "bad search space options first".

Reply:
You are always two steps in front of us 😀. Makes sense, thanks you.

Reply:
I just noticed this conversation, see my comment below.